lstchain/image/cleaning/cleaning.py

```diff
@@ -0,0 +1,136 @@
+        pixel values
+    interleaved_pedestal_thresh: array
+        additional picture threshold calculated using interleaved events
+    picture_thresh: float or array
+        threshold above which all pixels are retained
+    boundary_thresh: float or array
+        threshold above which pixels are retained if they have a neighbor
+        already above the picture_thresh
+    keep_isolated_pixels: bool
+        If True, pixels above the picture threshold will be included always,
+        if not they are only included if a neighbor is in the picture or
+        boundary
+    min_number_picture_neighbors: int
+        A picture pixel survives cleaning only if it has at least this number
+        of picture neighbors. This has no effect in case keep_isolated_pixels is True
+    Returns
+    -------
+    A boolean mask of *clean* pixels.  To get a zero-suppressed image and pixel
+    list, use `image[mask], geom.pix_id[mask]`, or to keep the same
+    image size and just set unclean pixels to 0 or similar, use
+    `image[~mask] = 0`
+    """
+
+    pixels_above_picture = np.logical_and(image>= picture_thresh,
+                                          image >= interleaved_pedestal_thresh)
+
+    if keep_isolated_pixels or min_number_picture_neighbors == 0:
+        pixels_in_picture = pixels_above_picture
+    else:
+        # Require at least min_number_picture_neighbors. Otherwise, the pixel
+        #  is not selected
+        number_of_neighbors_above_picture = geom.neighbor_matrix_sparse.dot(
+            pixels_above_picture.view(np.byte)
+        )
+        pixels_in_picture = pixels_above_picture & (
+            number_of_neighbors_above_picture >= min_number_picture_neighbors
+        )
+
+    # by broadcasting together pixels_in_picture (1d) with the neighbor
+    # matrix (2d), we find all pixels that are above the boundary threshold
+    # AND have any neighbor that is in the picture
+    pixels_above_boundary = image >= boundary_thresh
+    pixels_with_picture_neighbors = geom.neighbor_matrix_sparse.dot(pixels_in_picture)
+    if keep_isolated_pixels:
+        return (
+            pixels_above_boundary & pixels_with_picture_neighbors
+        ) | pixels_in_picture
+    else:
+        pixels_with_boundary_neighbors = geom.neighbor_matrix_sparse.dot(
+            pixels_above_boundary
+        )
+        return (pixels_above_boundary & pixels_with_picture_neighbors) | (
+            pixels_in_picture & pixels_with_boundary_neighbors
+)
+
+def get_bias_and_rms(dl1_file):
+    """
+    Function to extract bias and rms from interleaved events from dl1 file.
+    Parameters
+    ----------
+    input_filename: str
+        path to dl1 file
+    Returns
+    -------
+    bias, rms: np.ndarray, np.ndarray
+        bias and rms in p.e.
+    """
+    f = tables.open_file(dl1_file)
+    ped = f.root[dl1_params_tel_mon_ped_key]
+    ped_charge_mean = np.array(ped.cols.charge_mean)
+    ped_charge_rms = np.array(ped.cols.charge_std)
```

Comment:
For clarity, better use always std or std dev, not rms

Reply:
I don't understand this, ped.cols.charge_mean and charge_std do not contain just the interleaved event info, right @FrancaCassol ? They contain also (in row 0) the values from the dedicated calibration file, potentially taken way before the current subrun, and hence with different star positions.

Reply:
Yes, in row 0 we have pedestal for calibration run, in row 1 we have pedestal from interleaved events (at least that's how I understand it )

Reply:
Ok, but where do you select row 1?

Reply:
Hi @moralejo 
I think Paweł selects those here:
https://github.com/cta-observatory/cta-lstchain/blob/601684f5390210b7c6de49e4bfbf3ef60f29e269/lstchain/image/cleaning/cleaning.py#L136

as far as I remember what @FrancaCassol is doing in calculation of those pedestal updates is to use all the interleaved events of a given subrun to compute a single interleaved update of pedestals, so then you should have always exactly two values: original from dedicated calibration/pedestal run, and second (only one) from interleaveds. If we ever change this and could have 2 values of interleaveds in one subrun this code would need to be updated, but I think there is no point in doing it now. There is another problem, if the subrun is very short and have no interleaved pedestals for the update. I'm not sure if e.g. if you have 20 pedestal events if the values are computed at all (and have huge uncertainties) or not. In both cases it would be probably simplest to use the ped/cal run values then (it would either way not make much difference since those would be just small fractions of the whole dataset

Reply:
Ok, thanks a lot, it is clear now.
About short subruns, I agree it is not a big deal because they are a small fraction of data. But I would not be too strict... already with 10 pedestal events (collected in just 0.1 s) one should be able to tell a pixel containing a bright star from one with no star. And in any case, if the std dev has a large error, we may again argue that anyway it is few events that will be affected.
Eventually one can use also (properly treated) cosmics to improve the precision of pedestal std dev's.

Reply:
I was more thinking about regular, non-star-affected pixels that by a fluctuation will get a higher pedestal RMS and thus we would start increasing their threshold without any need.
I made a quick check with toy MC, Assuming that you get RMS of 1 phe and that you have 20 pedestal measurements the accuracy on pedestal bias is 0.22 phe (1 phe / sqrt(20) ), which if you have an actual values of bias of ~ 2phe would end up with ~10% accuracy. In the case of RMS of 1 phe , those 20 events would produce uncertainty of it of 0.16 phe (i.e. 16%, there was also a formula on the accuracy of sigma parameter from N measurements, but I do not remember it :-) ), so well, not that bad, but it is also enough to start slightly increasing thresholds of some pixels, so very short runs can be indeed affected, but probably only slightly.

Reply:
Probably better than completely omitting the anti-noisy pixels procedure. 

I don't know what the Travis CI problem is, that should also be addressed. And once we have a working dl1b script we can also arrange with @morcuended for testing it in La Palma with some full pet runs (2011 & 2012 for instance)

Reply:
Regarding Travis:
```
ERROR: Could not build wheels for ctapipe which use PEP 517 and cannot be installed directly
The command "pip install \
  https://github.com/cta-observatory/ctapipe/archive/$CTAPIPE_VERSION.tar.gz \
  https://github.com/cta-observatory/ctapipe_io_lst/archive/$CTAPIPE_IO_LST_VERSION.tar.gz \
  pytest-cov codecov
  " failed and exited with 1 during .
```
Since GitHub actions are already in place why don't we get rid of Travis already?

> And once we have a working dl1b script we can also arrange with @morcuended for testing it in La Palma with some full pet runs (2011 & 2012 for instance)

Sure, let me know the details whenever is done. I remind you that PR #543 could be merged already. The new cleaning feature using pedestal rms can be implemented in a separate PR (?).

Reply:
> Since GitHub actions are already in place why don't we get rid of Travis already?

I guess it'd be just merging #578